annotation/tests/test_annotations.py

```diff
@@ -83,3 +172,142 @@ def test_row_to_dict(row, expected_dictionary):
     result = row_to_dict(row)
     print(result)
     assert result == expected_dictionary
+
+
+@pytest.mark.unittest
+@pytest.mark.parametrize(
+    [
+        "s3_prefix",
+        "bucket_name",
+        "expected_string",
+    ],
+    [
+        (
+            "S3_test",
+            "bucket_test",
+            "S3_test-bucket_test",
+        ),
+        (
+            None,
+            "bucket_test",
+            "bucket_test",
+        ),
+    ],
+)
+def test_convert_bucket_name_if_s3prefix(
+    monkeypatch,
+    s3_prefix,
+    bucket_name,
+    expected_string,
+):
+    monkeypatch.setattr("annotation.annotations.main.S3_PREFIX", s3_prefix)
```

Comment:
Let's use unittest.mock patch decorator/context manager for such cases instead of monkeypatch

Reply:
https://docs.pytest.org/en/stable/how-to/fixtures.html#yield-fixtures-recommended

Reply:
https://docs.python.org/3/library/unittest.mock.html#the-patchers